### ISSUE TITLE
@include{doc}

### ISSUE THREAD

**Issue by AndreasMatthias** [doxygen/doxygen#10702]
File `bar.cpp`:
```cpp
///
/// See @ref XXX.
///
class Bar {
};

/// @include{doc} page.dox
```

File `page.dox`:
```
@page XXX Additional page
/// This should be
/// a separate page.
```

HTML output is fine, nevertheless file `page.dox` looks kind of weird. Note that the first line is *not* commented while all other lines are commented. Any other combination (all line commented, all lines uncommented, etc.) will produce some garbage.

This might be related to #10700.

**Comment by albert-github** [comment]
You mean that it is a bit weird to have the `///` in the file page.dox
This is due to the handling of the single line `///` type of comments. When we look with `doxygen -d commentcbv` we see:
```
/**
 *  See @ref XXX.
 *  */
class Bar {
};

///  \ilinebr \ifile ".../page.dox" \iline 1 \ilinebr @page XXX Additional page
/// This should be
/// a separate page.
 \ilinebr \ifile ".../bar.cpp" \iline 7
```
which shows also another problem (i.e. the line ` \ilinebr \ifile ".../bar.cpp" \iline 7`).

Creating a multiline `///` comment will solve this problem, so using:
```
/// @include{doc} page.dox
///
```

and at that moment the `/// ` can and should be removed by the user from page.dox

@doxygen should we classify this as a bug or usage?

**Comment by albert-github** [comment]
@AndreasMatthias 

I was able to reproduce the problem (see https://github.com/doxygen/doxygen/issues/10702#issuecomment-1971532793) bit to be sure we are talking about the same problem always:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).
- Please also specify the **full** doxygen version used (`doxygen -v`).

**Comment by AndreasMatthias** [comment]
Oh, now I see. Thanks. The problem was that I didn't know that there was a difference between **single line comments**
```
/// @include{doc} page.dox
```
and **multi-line comments**
```
/// @include{doc} page.dox
///
```
Now it makes more sense why I have to **comment** or **not comment** certain lines in `page.dox`.

Since I was using a **single-line comment**, I had to write `page.dox` like this:
```
@page XXX Additional page
/// This should be
/// a separate page.
```
But if I used a **multi-line comment**, then `page.dox` would had to be like this:

```
@page XXX Additional page
This should be
a separate page.
```
Now I understand. And I guess the output of doxygen is ok. It just wasn't intuitive for me.

**Comment by AndreasMatthias** [comment]
I'm running:
```
$ doxygen -v
1.11.0 (50d177669564432d8cfcab4d146bb19f024aec7e)
```

**Comment by AndreasMatthias** [comment]
Here is a complete example.

[test_doxy.tar.gz](https://github.com/doxygen/doxygen/files/14451523/test_doxy.tar.gz)

The difference of `page1.dox` and `page2.dox` seems weird at first. But once you understand the difference between **single line comments** and **multi line comments**, it makes perfect sense.

**Comment by doxygen** [comment]
@AndreasMatthias Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by AndreasMatthias** [comment]
Tested with commit cdafc3d25:

My real world examples are fine. But testing it further, there are still some unexpected results. 

See this example.
[test_doxy.tar.gz](https://github.com/doxygen/doxygen/files/14485417/test_doxy.tar.gz)
I'm not sure what the supposed output should look like. I added some (more or less) random stuff and ended up with 4 pages, none of which looked like I expected it to be. Especially this number on page 4. (A line counter? Number depends on the number of lines in page4.dox)

**Comment by albert-github** [comment]
For the given example I get indeed the warning:
```
page4.dox:4: warning: included file 'page3.dox\iline' is not found. Check your EXAMPLE_PATH
```
problem looks like to be introduced in the "commentscan" phase.

The results on pages 2 and 4 don't look like how they should be.

**Comment by albert-github** [comment]
I get the impression that the problem reported in https://github.com/doxygen/doxygen/issues/10702#issuecomment-1977207135 is fixed with commit [8bc3d9d](https://github.com/doxygen/doxygen/commit/8bc3d9dc0b3f90c881bcc2db7652c7063039a6c5)

@AndreasMatthias Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by AndreasMatthias** [comment]
Well, as I said, I'm not sure what the "correct" output should look like. Take for example page-4:
![page-4](https://github.com/doxygen/doxygen/assets/1072202/406acc30-4cb1-47f9-85c5-e83a9cdf9c4b)

Page 4 contains the source code block of page-3. Similar for page 2. While the source code block on page-1 and page-3 is missing. Is this the expected behavior?

**Comment by albert-github** [comment]
The problem here is that for the code snippet:
```
/// @include{doc} page3.dox

/// @include page3.dox

/// @include{doc} page4.dox

/// @include page4.dox
```
we see that the first comment block with `@include{doc} page3.dox` is ended by the empty line and the following comment block starts with `@include page3.dox` and ends with the empty line. The `@include page3.dox` block is in the middle of nowhere and will by picked up with the next "entity" command i.e. here the comment block with `@include{doc} page4.dox`.

So it looks a bit illogical but I think this is intended and can be prevented by adding e.g the `@include page3.dox` block to the `@include{doc} page3.dox` by either removing the empty line in between or changing that empty line into a comment line with just `///` on it.

**Comment by doxygen** [comment]
I agree with Albert.

`/// @include page3.dox` starts a brief description, but then the `@include` command will immediately turn the comment into a detailed description.

`/// @include{doc} page4.dox` starts a page with a brief description. 

These two are then joined together and appear in reverse order as brief description is always put before the detailed description.

**Comment by AndreasMatthias** [comment]
@albert-github 
@doxygen 

Ok, I see. Thank you for the explanation and the quick bug fixes.

I guess this issue is fixed, now. Thanks.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
